src/main/java/com/onehouse/metadata_extractor/ActiveTimelineInstantBatcher.java

```diff
@@ -107,9 +135,23 @@ public List<List<File>> createBatches(List<File> instants, int maxBatchSize) {
           currentBatch.clear();
           currentBatch.addAll(sortedInstants.subList(index, index + groupSize));
         }
-      } else {
-        // Instants are not related; add what we have and stop processing
-        shouldStopIteration = true;
+      } else if (!shouldStopIteration) {
```

Comment:
we should do this even if shouldStopIteration is true right? as shouldStopIteration is true when there are commits which are incomplete